### PR TITLE
Add basic ci

### PR DESCRIPTION
**Pull request by jeising**
Add CI for:
- colcon build, test and test results (.github/workflows/main.yml)
- iceoryx build (.github/workflows/iceoryx.yml)

Resolves: #10

### CODE REVIEW

**Comment by jeising** [comment]
@michael-poehnl: There seems to be some inconsistency with uncrustify on current master.
see: [ci test results](https://github.com/jeising/rmw_iceoryx/runs/536658722?check_suite_focus=true)

**Comment by Karsten1987** [comment]
Thanks @jeising. I think it's a great starting point. We could also consider adding the linters in a separate action. See here for examples: https://github.com/ros-controls/ros2_controllers/pull/33/files

As or the current uncrustify failures, they are going to be addressed in #15

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        with:
+          repository: eclipse/iceoryx
+          path: src/iceoryx
+          ref: master
+      - name: Trigger Build
+        run: "source /opt/ros/${{ matrix.ros }}/setup.bash && colcon build"
+      - name: Execute Tests
+        run: "source /opt/ros/${{ matrix.ros }}/setup.bash && colcon test"
+      - name: Show Test Results
+        run: "source /opt/ros/${{ matrix.ros }}/setup.bash && colcon test-result --verbose"
```
Comment (Karsten1987) [inline .github/workflows/main.yml]: new line missing

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+name: CI
```
Comment (Karsten1987) [inline .github/workflows/main.yml]: maybe a bit more descriptive name for it?

```suggestion
name: CI rmw_iceoryx
```

**Comment by jeising** [comment]
I have no write access so @Karsten1987 feel free to merge it.

**Comment by Karsten1987** [comment]
Looking at #15 it might be worth to limit the amount of jobs of `push` to 
```
push:
    branches:
      - master
```

**Comment by jeising** [comment]
Lets put it there as soon as we need it :)

**Pull request merged**